packages/codegen-ui/lib/utils/form-to-component/map-form-definition-to-component.ts

```diff
@@ -98,55 +98,84 @@ const ctaButtonConfig = (): StudioComponentChild => {
         value: '1rem',
       },
     },
-    children: [
-      {
-        componentType: 'Button',
-        name: 'CancelButton',
-        properties: {
-          children: {
-            value: 'Cancel',
-          },
-          type: {
-            value: 'button',
-          },
+    children: [],
+  };
+
+  if (formDefinition.buttons.buttonConfigs.cancel) {
+    CTAComponent.children?.push({
+      componentType: 'Button',
+      name: 'CancelButton',
+      properties: {
+        children: {
+          value: formDefinition.buttons.buttonConfigs.cancel.props.children,
+        },
+        type: {
+          value: 'button',
         },
       },
-      {
-        componentType: 'Flex',
-        name: 'SubmitAndResetFlex',
-        properties: {},
-        children: [
-          {
-            componentType: 'Button',
-            name: 'ClearButton',
-            properties: {
-              children: {
-                value: 'Clear',
-              },
-              type: {
-                value: 'reset',
-              },
-            },
-          },
-          {
-            componentType: 'Button',
-            name: 'SubmitButton',
-            properties: {
-              children: {
-                value: 'Submit',
-              },
-              type: {
-                value: 'submit',
-              },
-              variation: {
-                value: 'primary',
-              },
-            },
-          },
-        ],
-      },
-    ],
+    });
+  }
+
+  const rightAlignCTA: StudioComponentChild = {
+    componentType: 'Flex',
+    name: 'SubmitAndResetFlex',
+    properties: {},
+    children: [],
   };
+
+  if (formDefinition.buttons.buttonConfigs.clear && rightAlignCTA.children) {
+    rightAlignCTA.children.push({
+      componentType: 'Button',
+      name: 'ClearButton',
+      properties: {
+        children: {
+          value: formDefinition.buttons.buttonConfigs.clear.props.children,
+        },
+        type: {
+          value: 'reset',
+        },
+      },
+    });
+  }
+
+  if (formDefinition.buttons.buttonConfigs.submit) {
+    rightAlignCTA.children?.push({
+      componentType: 'Button',
+      name: 'SubmitButton',
+      properties: {
+        children: {
+          value: formDefinition.buttons.buttonConfigs.submit.props.children,
+        },
+        type: {
+          value: 'submit',
+        },
+        variation: {
+          value: 'primary',
+        },
+      },
+    });
+  }
+
+  CTAComponent.children?.push(rightAlignCTA);
+
+  return CTAComponent;
+};
+
+const renderChildren = (name: string, formDefinition: FormDefinition) => {
+  switch (formDefinition.buttons.position) {
```

Comment:
What do you think about creating a file for button-related helpers?

Reply:
This is a good suggestion, I don't want to bloat this file - do you want it added to this PR or in a follow up?